modules/sdk/src/main/scala/org/tessellation/sdk/domain/rewards/Rewards.scala

```diff
@@ -7,13 +7,15 @@ import org.tessellation.schema.balance.Balance
 import org.tessellation.schema.snapshot.{IncrementalSnapshot, StateProof}
 import org.tessellation.schema.transaction.{RewardTransaction, Transaction}
 import org.tessellation.sdk.infrastructure.consensus.trigger.ConsensusTrigger
+import org.tessellation.sdk.snapshot.currency.CurrencySnapshotEvent
 import org.tessellation.security.signature.Signed
 
 trait Rewards[F[_], P <: StateProof, S <: IncrementalSnapshot[P]] {
   def distribute(
     lastArtifact: Signed[S],
     lastBalances: SortedMap[Address, Balance],
     acceptedTransactions: SortedSet[Signed[Transaction]],
-    trigger: ConsensusTrigger
+    trigger: ConsensusTrigger,
+    events: Set[CurrencySnapshotEvent]
```

Comment:
looks like it leaks the implementation to the declaration. at this point we know nothing about `S` being a global snapshot or currency snapshot. I think you should parametrize `events` and pass global snapshot events on GL0 and currency snapshot events on Currency L0

Reply:
@ryle-goehausen ☝️ FYI